engine/src/main/java/com/datatorrent/stram/codec/DefaultStatefulStreamCodec.java

```diff
@@ -63,10 +64,11 @@ public DefaultStatefulStreamCodec()
     classResolver = (ClassResolver)getClassResolver();
     this.pairs = classResolver.pairs;
     classResolver.init();
+    dataStatePair = new DataStatePair();
```

Comment:
This makes the assumption that caller will not delay the processing of the return value, which is true, but it breaks the assumption that every tuple is a distinct object.